### PR TITLE
Use db type jsonb to support dot notation filters

### PR DESCRIPTION
**Pull request by ewrayjohnson**
Using dot notation in query filters does not work because the database column type for object and embedded models is string.  This fixes the problem by changed the type to jsonb.

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by dhmlau** [comment]
@ewrayjohnson, thanks for your PR. Could you please add some tests to verify your code, and also rebase your branch? Thanks.

File: lib/migration.js [hunk 1]
```diff
@@ -479,9 +479,9 @@ function mixinMigration(PostgreSQL) {
     const p = propertyDefinition;
     switch (p.type.name) {
       default:
-      case 'String':
       case 'JSON':
-        return 'TEXT';
+        return 'JSONB';
```
Comment (bajtos) [inline lib/migration.js]: What is the impact of this change on existing application? Have you considered introducing a new `p.type.name` called `JSONB`, that will allow app developers using `JSON` to preserve the current behavior, and let them opt into the new JSONB-based storage?

File: lib/migration.js [hunk 2]
```diff
@@ -783,8 +783,7 @@ function mixinMigration(PostgreSQL) {
           if (identical && siKeys.length > 1) {
             if (siKeys.length !== i.keys.length) {
               // lengths differ, obviously non-matching
-              // ??? orderMatched is not defined and not used anywhere else
-              // orderMatched = false;
+              identical = false;
```
Comment (bajtos) [inline lib/migration.js]: This change looks unrelated to the other change you are making above. What scenario are you trying to fix with this change?

**Comment by ewrayjohnson** [comment]
I am working on creating tests to show the bug and confirm the fix.
